docs/Architecture.md

```diff
@@ -1,42 +1,55 @@
 
-The API is responsible for converting the external protocol (currently gRPC with grpc-gateway) to entities, invoking the necessary service, and converting the response back.
+> ⚠ Note: The requests that Maestro receives in the Rooms API are documented in [this proto file](https://github.com/topfreegames/protos/blob/master/maestro/grpc/protobuf/events.proto).
 
-### Workers
-Maestro now provides a form of extending and adding more workers if they follow a worker interface. This was done to have the same pattern applied to the operation execution worker and runtime watcher. Workers are managed by the "WorkersManager," which starts an instance for each scheduler present on the storage. With this model, the implementation doesn't have to deal with multiple tenants (schedulers).
+![Rooms API IMAGE](./diagrams/Architecture-Rooms-API.jpg)
 
-Workers are not inside the service package, although they hold business logic about execution and flows. They were isolated since workers are a kind of service that has to implement an interface. Besides this difference, workers should be treated like core services.
+#### Operation Execution Worker
+
+> ⚠ Note: In Maestro a worker is a collection of routines that executes a flow related to one and only one **Scheduler** each.
+
+Operation Execution Worker creates for each **Scheduler** a thread that executes operations enqueued in the related **Scheduler** operation queue. So in this way became possible to track the events that happened and change a certain **Scheduler** in a healthier way.
+
+The Operations could be:
+- Create Scheduler: creates a **Scheduler**
+- New Scheduler Version: updates a **Scheduler**
+- Switch Active Version: change the current **Scheduler** version that is running. It uses the versions to know if the Game Rooms need to be changed
+- Add Rooms: adds a number of Game Rooms in a **Scheduler**
+- Remove Rooms: removes a number of Game Rooms in a **Scheduler**
+
+![Operation Execution Worker IMAGE](./diagrams/Architecture-Operation-Execution-Worker.jpg)
+
+#### Runtime Watcher Worker
+
+> ⚠ Note: In Maestro a worker is a collection of routines that executes a flow related to one and only one **Scheduler** each.
+
+Runtime Watcher Worker listens to runtime events related to the **Scheduler** and reflects the changes in the Maestro. Currently, it listens for Game Rooms creation and deletion.
+
+![Runtime Watcher Worker IMAGE](./diagrams/Architecture-Runtime-Watcher-Worker.jpg)
+
+#### Metrics Reporter Worker
+
+> ⚠ Note: In Maestro a worker is a collection of routines that executes a flow related to one and only one **Scheduler** each.
+
+From time to time Metrics Reporter Worker watch runtime to report metrics from them, such as the number of game rooms instances that are `ready`, `pending`, `error`, `unknown`, or `terminating` status. As well it watches from Game Rooms storage its status that could be `ready`, `pending`, `error`, `occupied`, `terminating`, or `unready`.
```

Comment:
```suggestion
From time to time Metrics Reporter Worker watch runtime to report metrics from them, such as the number of game rooms instances that are `ready`, `pending`, `error`, `unknown`, or `terminating` status. As well it watches from Game Rooms storage its status that could be `ready`, `pending`, `error`, `occupied`, `terminating`, or `unready`.

This module is optional since you don't need it for any specific functionalities of the application.
```